fs-sim/src/System/FS/Sim/MockFS.hs

```diff
@@ -491,7 +495,7 @@ hOpen fp openMode = do
           , fsErrorStack  = prettyCallStack
           , fsLimitation  = True
           }
-      when (openMode == ReadMode) $ void $
+      when (openMode == ReadMode || assumedExistance openMode) $ void $
         checkFsTree $ FS.getFile fp (mockFiles fs)
```

Comment:
If we make the change `allowExisting ReadMode = MustExist`, then I think this check can go away completely. `openFile` should then already throw the correct error